Library/OcCryptoLib/X64/sha512-avx.nasm

```diff
@@ -0,0 +1,532 @@
+; @file
+; Copyright (C) 2021, ISP RAS. All rights reserved.
+;
+; All rights reserved.
+;
+; This program and the accompanying materials
+; are licensed and made available under the terms and conditions of the BSD License
+; which accompanies this distribution.  The full text of the license may be found at
+; http://opensource.org/licenses/bsd-license.php
+;
+; THE PROGRAM IS DISTRIBUTED UNDER THE BSD LICENSE ON AN "AS IS" BASIS,
+; WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
+;
+; #######################################################################
+;
+;  This code is described in an Intel White-Paper:
+;  "Fast SHA-512 Implementations on Intel Architecture Processors"
+;
+; ########################################################################
+; ### Binary Data
+BITS 64
+
+section .rodata
+align 16
+; Mask for byte-swapping a couple of qwords in an XMM register using (v)pshufb.
+XMM_QWORD_BSWAP:
+	dq 0x0001020304050607,0x08090a0b0c0d0e0f
+
+align 64
+; K[t] used in SHA512 hashing
+K512:
+	dq 0x428a2f98d728ae22,0x7137449123ef65cd
```

Comment:
These constants are already made available in SHA512_K of the C code. Since AVX version needs custom alignment, in my opinion, drop the C version and make this symbol globally accessible for C code to use. Doing this will also reduce the resulting code size.